blackjax/kernels.py

```diff
@@ -672,41 +672,41 @@ def window_adaptation(
     A function that returns the last chain state and a sampling kernel with the tuned parameter values from an initial state.
 
     """
-
     kernel = algorithm.kernel()
 
-    def kernel_factory(step_size: float, inverse_mass_matrix: Array):
-        def kernel_fn(rng_key, state):
-            return kernel(
-                rng_key,
-                state,
-                logprob_fn,
-                step_size,
-                inverse_mass_matrix,
-                logprob_grad_fn=logprob_grad_fn,
-                **parameters,
-            )
-
-        return kernel_fn
-
     schedule = adaptation.window_adaptation.schedule(num_steps)
     init, update, final = adaptation.window_adaptation.base(
-        kernel_factory,
         is_mass_matrix_diagonal,
-        target_acceptance_rate=target_acceptance_rate,
+        initial_step_size,
+        target_acceptance_rate,
     )
 
     def one_step(carry, xs):
         _, rng_key, adaptation_stage = xs
         state, adaptation_state = carry
-        state, adaptation_state, info = update(
-            rng_key, state, adaptation_state, adaptation_stage
+
+        new_state, info = kernel(
```

Comment:
I am getting confuse here for a moment because we have too many things call kernel - could you add docstring or use `mcmc_one_step` (or `mutation`/`state_transition` etc) here?

Reply:
Yes, kernel is not a very informative name.